diff --git a/python/postprocessing/framework/postprocessor.py b/python/postprocessing/framework/postprocessor.py
index 1c75036..ceaf437 100755
--- a/python/postprocessing/framework/postprocessor.py
+++ b/python/postprocessing/framework/postprocessor.py
@@ -144,6 +144,7 @@ class PostProcessor:
 
         fullClone = (len(self.modules) == 0)
         outFileNames = []
+        outFileNameshadd = ""
         t0 = time.time()
         totEntriesRead = 0
         for fname in self.inputFiles:
@@ -199,6 +200,8 @@ class PostProcessor:
             if not self.noOut:
                 outFileName = os.path.join(self.outputDir, os.path.basename(
                     fname).replace(".root", outpostfix + ".root"))
+                outFileNameshadd = os.path.join(self.outputDir, os.path.basename(
+                    fname).replace(".root", outpostfix + "Hadd.root"))
                 outFile = ROOT.TFile.Open(
                     outFileName, "RECREATE", "", compressionLevel)
                 outFileNames.append(outFileName)
@@ -257,8 +260,18 @@ class PostProcessor:
         if self.haddFileName:
             haddnano = "./haddnano.py" if os.path.isfile(
                 "./haddnano.py") else "haddnano.py"
+            print("\n\n[postprocessor.py::INFO::] %s %s %s" %
+                  (haddnano, outFileNameshadd," ".join(outFileNames)))
+            print ""
+            os.system('date')
+            startTime = time.clock()
+            print ""
             os.system("%s %s %s" %
-                      (haddnano, self.haddFileName, " ".join(outFileNames)))
+                      (haddnano, outFileNameshadd, " ".join(outFileNames)))
+            print ""
+            os.system('date')
+            print "Time taken for hadd is %s s"%(time.clock() - startTime)
+            print ""
         if self.jobReport:
             self.jobReport.addOutputFile(self.haddFileName)
             self.jobReport.save()
